modules/sequence/src/main/scala/lucuma/odb/data/OneOf3.scala

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.data
+
+import cats.Eq
+import cats.syntax.either.*
+import cats.syntax.eq.*
+import cats.syntax.option.*
+
+/**
+ * Like an Either, but with three different cases and no error / success
+ * connotation or right bias.
+ */
+sealed trait OneOf3[+A, +B, +C]:
```

Comment:
So, I need to stream three different things in timestamp order: step records, visit starts and sequence events.  I use this to produce a single `Stream` that contains one of the 3 options.  Maybe there is something standard for this in `cats` or maybe this is just dumb?